packages/popover/src/Popover.styles.ts

```diff
@@ -78,33 +95,91 @@ const transformOriginStyles: Record<ExtendedPlacement, string> = {
   `,
 };
 
-const getTransformStyles = (placement: ExtendedPlacement, spacing: number) =>
-  cx({
-    [css`
+const getClosedStyles = (placement: ExtendedPlacement, spacing: number) => {
+  if (placement.startsWith('top')) {
```

Comment:
I think breaking `placement` into its constituent `align` and `justify`, and then using a `switch` here would make this a bit more readable, and would likely be easier to test